pkg/controller/logstorage/esgateway.go

```diff
@@ -47,7 +47,9 @@ func (r *ReconcileLogStorage) createEsGateway(
 		return reconcile.Result{}, false, nil
 	}
 
-	kubeControllersGatewaySecret, kubeControllersVerificationSecret, kubeControllersSecureUserSecret, err := common.CreateKubeControllersSecrets(ctx, esAdminUserSecret, r.client)
+	esAdminUserName := "elastic"
```

Comment:
Shouldn't the username come from the secret? Shouldn't there always be just one key in the secret, which is the username? (And the value being the password?)

Reply:
The key is `elastic` so the only way that I can think of to access the password is to do `esAdminUserSecret.Data["elastic"]`, however, for multi-tenancy, the key can't be `elastic` since that user is reserved for the cloud team and instead it will be `tigera-mgmt`. This is why I made this change so that I can instead go `esAdminUserSecret.Data[esAdminUserName]` and pass in whatever the key should be based on if the Operator is setting up a single or multi-tenant env.

Reply:
Actually I think I misunderstood what you're saying, I should just extract the key from the `esAdminUserSecret.Data` map and use that. I'll update the PR

Reply:
Yes, that is what I am thinking.